plugins/postcss-logical/src/lib/transform-inset.ts

```diff
@@ -20,24 +21,30 @@ export function transformInset(
 			'left': '',
 		};
 
+		if (hasVariables && values.length !== 4) {
+			// A variable can "resolve" to multiple separate values.
+			// This makes it impossible to statically map tokens to longhand properties.
+			return;
+		}
+
 		if (values.length === 1) {
 			newRules.top = valueParser.stringify(values[0]);
-			newRules.right = valueParser.stringify(values[0]);
```

Comment:
`valueParser.stringify` can be slow and expensive.
Better to re-use work.

Reply:
Nicely done! I also think this is more clear!